File: java/src/ch2/ChildClass.java
package ch2;

public class ChildClass extends ParentClass {

    /*
    public ChildClass(int field, int field2) {
        super(field, field2);
    } */

    public ChildClass(int field) {
        super(field);
    }

    void printChild () {
        System.out.println("ChildClass: " + field);
    }
}

File: c/ch1/child_class.c
#include "child_class.h"
#include "parent_class.h"

#include <stdlib.h>
#include <stdio.h>

child_class *init_child_class (int field)
{
    child_class *tmp = (child_class *)malloc(sizeof(child_class));
    tmp->base_obj = init_parent_class(field);
    tmp->base_obj->derived_obj = tmp;

    return tmp;
}

void print_child (child_class *obj)
{
    printf("I'm child - field = %d\n", obj->base_obj->field);
}

void free_child_class (child_class *obj)
{
    free(obj->base_obj);
    free(obj);
}
